Unstall the Forever stalled MDM PR

SaitejaUtpala opened the pull request:
This is an (optimistic )attempt to un-stall the #962 by breaking into series of smaller PRs. As first this just includes mdm algorithm and corresponding test.

nguigs commented on geomstats/learning/mdm.py:
```diff
@@ -0,0 +1,114 @@
+        frechet_means = []
+        for c in range(self.n_clusters):
+            data_class = self.split_data_in_classes(X, y, c)
+            # frechet_means.append(mean_estimator.fit(data_class).estimate_[0])
```
left-over commented lign

nguigs commented on geomstats/learning/mdm.py:
```diff
@@ -0,0 +1,114 @@
+
+        Parameters
+        ----------
+        X : array-like, shape=[n_samples, n_features]
+                              if point_type='vector'
+                              shape=[n_samples, n_features, n_features]
+                              if point_type='matrix'
```
This may not be the case in the rest of the code base, but I don't think we should use `n_features`, but `dim` instead, as this carries more the geometric attribute of the manifold: features are independent parameters, shaping a vector space.
For the case `point_type='matrix'`, we usually use `[n, n]` with no real meaning

nguigs reviewed (approved):
Great to split the PR into smaller ones! Thanks for handlind this!

ninamiolane commented:
Why is code coverage decreasing? 
"Merging #1003 (909b077) into master (c5304ce) will decrease coverage by 11.72%."

Also, any idea why the tests fail?

ninamiolane commented on geomstats/learning/mdm.py:
```diff
@@ -0,0 +1,114 @@
+
+
+class RiemannianMinimumDistanceToMeanClassifier:
+    """
```
Docstring formatting conventions:
https://geomstats.github.io/contributing.html#docstring-examples

ninamiolane commented:
Awesome idea indeed!! Thank you @SaitejaUtpala 🙏

SaitejaUtpala commented:
@ninamiolane @nguigs . I have fixed doc string issues. one thing is that there are these weird errors on the **notebooks/usecase_emg_sign_classification_in_spd_manifold.ipynb** at the line  `keras.wrappers.scikit_learn import KerasClassifier`  and Error is `AttributeError: module 'keras.utils.generic_utils' has no attribute 'populate_dict_with_module_objects'` . I didn't have error before, It was throwing this error suddently.  This error also started popping at  #1005

nguigs commented:
> @ninamiolane @nguigs . I have fixed doc string issues. one thing is that there are these weird errors on the **notebooks/usecase_emg_sign_classification_in_spd_manifold.ipynb** at the line `keras.wrappers.scikit_learn import KerasClassifier` and Error is `AttributeError: module 'keras.utils.generic_utils' has no attribute 'populate_dict_with_module_objects'` . I didn't have error before, It was throwing this error suddently. This error also started popping at #1005

This is very weird indeed, I changed the import to `tensorflow.keras` to avoid mismatch between versions. It worked and is now merged into master so rebasing should work for you

SaitejaUtpala commented:
> > @ninamiolane @nguigs . I have fixed doc string issues. one thing is that there are these weird errors on the **notebooks/usecase_emg_sign_classification_in_spd_manifold.ipynb** at the line `keras.wrappers.scikit_learn import KerasClassifier` and Error is `AttributeError: module 'keras.utils.generic_utils' has no attribute 'populate_dict_with_module_objects'` . I didn't have error before, It was throwing this error suddently. This error also started popping at #1005
> 
> This is very weird indeed, I changed the import to `tensorflow.keras` to avoid mismatch between versions. It worked and is now merged into master so rebasing should work for you

https://stackoverflow.com/questions/67049352/attributeerror-module-keras-utils-generic-utils-has-no-attribute-populate-di . Yup it is indeed due to version mismatch.

ninamiolane reviewed (approved):
Yooohooo!! 🎉

SaitejaUtpala commented:
@ninamiolane  @nguigs . I rewrote the MDM completely to make sure it works pytorch and tensorflow backend as well. Last time it was working only with numpy backend
